system/jlib/jsocket.cpp

```diff
@@ -368,6 +368,71 @@ enum SOCKETMODE { sm_tcp_server, sm_tcp, sm_udp_server, sm_udp, sm_multicast_ser
 # endif
 #endif
 
+static CriticalSection queryKACS;
+
+enum UseKA { UNINIT, DISABLED, ENABLED };
+static UseKA doKeepAlive = UNINIT;
+static int keepAliveTime = 7200;
+static int keepAliveIntvl = 75;
+static int keepAliveProbes = 9;
+
+/*
+<Software>
+  <Globals>
+    <expert>
+      <keepalive time="200" interval="75" probes="9"/>
+    </expert>
+  </Globals>
+
+global:
+  expert:
+    keepalive:
+      time: 200
+      interval: 75
+      probes: 9
+*/
+
+static bool queryKeepAlive(int &time, int &intvl, int &probes)
```

Comment:
pedantic: shouldn't these really be unsigned's, since -ve would be invalid ?

Reply:
did you agree with this one?

Reply:
left as ints so we can easily use -1 as not specified in config.

Reply:
@mckellyln - the idea or our convension for the 'Resolved conversation' button, is for it to be used by the reviewer only, so it helps the reviewer to track their comments: 
https://github.com/hpcc-systems/HPCC-Platform/blob/master/devdoc/CodeReviews.md#:~:text=Only%20the%20reviewer%20should%20mark%20issues%20as%20resolved%20using%20the%20Github%20resolve%20conversation%20button.